### ISSUE TITLE
Unphased VCF but can generate phased FASTA

### ISSUE THREAD

**Issue by ettorefedele** [whatshap/whatshap#354]
Hi everyone,

I have used whatshap phase to generate a phased VCF, but the variants contained in the output VCF remain unphased. However, when I try to use the bcftools consensus function to generate FASTA files for H1 and H2, it works OK and the results seem to be in line with what I can see in IGV. Can anyone help me understand why I manage to get the FASTA but I cannot phase the VCF? Many thanks.

**Comment by marcelm** [comment]
How did you find out that the VCF is not phased? Can you please run `whatshap stats` on the file and check the output, in particular the last "ALL chromosomes" section?

**Comment by ettorefedele** [comment]
Hi Marcel,

thanks for your reply. I opened the VCF in excel and it looked exactly like the input VCF (0/0, 0/1, 1/1). This is the output of whatshap stats:

---------------- ALL chromosomes (aggregated) ----------------
     Variants in VCF:      178
        Heterozygous:       90 (      80 SNVs)
              Phased:        0 (       0 SNVs)
            Unphased:       90 (not considered below)
          Singletons:        0 (not considered below)
              Blocks:        0

Block sizes (no. of variants)
   Median block size:         nan variants
  Average block size:         nan variants
       Largest block:        0    variants
      Smallest block:        0    variants

Block lengths (basepairs)
      Sum of lengths:        0    bp
 Median block length:         nan bp
Average block length:         nan bp
       Longest block:        0    bp
      Shortest block:        0    bp
           Block N50:      nan    bp



I thought the problem was the lack of sample ID (WARNING: Sample 'xxxx' not found in any BAM file.), so I used the  --ignore-read-groups option (I am looking at one sample only), but I still get the same problem. This is the output of whatshap stats with the newly generated VCF:

---------------- ALL chromosomes (aggregated) ----------------
     Variants in VCF:      181
        Heterozygous:      181 (     181 SNVs)
              Phased:        0 (       0 SNVs)
            Unphased:      181 (not considered below)
          Singletons:        0 (not considered below)
              Blocks:        0

Block sizes (no. of variants)
   Median block size:         nan variants
  Average block size:         nan variants
       Largest block:        0    variants
      Smallest block:        0    variants

Block lengths (basepairs)
      Sum of lengths:        0    bp
 Median block length:         nan bp
Average block length:         nan bp
       Longest block:        0    bp
      Shortest block:        0    bp
           Block N50:      nan    bp

I generated the input VCF using samtools mpileup | bcftools call. I have also tried with Freebayes but I spotted a few errors.
Thanks!

**Comment by marcelm** [comment]
Ok, so the VCF is indeed unphased. Which `whatshap phase` command-line options did you use? What does it output? (How many reads did it use etc.)

Regarding bcftools consensus, what do you mean with "the results seem to be in line with what I can see in IGV"? Do you refer to the variants you see in the reads?

**Comment by ettorefedele** [comment]
Thanks. This is the command-line I used:

whatshap phase -o phased.vcf --reference=Reference.fasta genotyped.vcf Sample_6_sorted.bam

and then 

whatshap phase -o phased.vcf --reference=Reference.fasta genotyped.vcf Sample_6_sorted.bam --ignore-read-groups

This is what I get for each marker I am looking at:

---- Processing individual sample
Using maximum coverage per sample of 15X
Number of variants skipped due to missing genotypes: 3
Number of remaining heterozygous variants: 0
Reading alignments for sample 'sample' and detecting alleles ...
WARNING: Sample 'sample' not found in any BAM/CRAM file.
Found 0 reads covering 0 variants
Kept 0 reads that cover at least two variants each
Reducing coverage to at most 15X by selecting most informative reads ...
Selected 0 reads covering 0 variants
Best-case phasing would result in 0 non-singleton phased blocks (0 in total)
... after read selection: 0 non-singleton phased blocks (0 in total)
Variants covered by at least one phase-informative read in at least one individual after read selection: 0
Phasing 1 sample by solving the MEC problem ...
MEC cost: 0
No. of phased blocks: 0
======== Writing VCF
Done writing VCF

Regarding IGV, yes - I refer to the variant I see in the reads. I am not sure this is important/relevant, but I am using a Reference.fasta file that is basically a list of targeted loci.  Thank you very much for your help!

**Comment by ettorefedele** [comment]
Yes - that was it. It works now. Thank you very much for your help and sorry for taking your time.
